File: backend-app/Lostpet/Domain/Models/Announcement.cs
using Domain.Models.Auth;
using Domain.Models.Enums;

namespace Domain.Models;

public class Announcement : BaseEntity
{
    public AnnouncementPetStatus PetStatus { get; set; }

    public string? Country { get; set; }

    public string? City { get; set; }

    public decimal? LastSeenLatitude { get; set; }

    public decimal? LastSeenLongitude { get; set; }

    public string? NearLandmark { get; set; }

    public DateTimeOffset? LastDateWhenSeen { get; set; }

    public string? ApproximateTime { get; set; }

    public string? PetDetails { get; set; }

    public bool IsPhonePublic { get; set; }

    public bool IsTelegramActive { get; set; }

    public bool IsActive { get; set; } = true;

    public AnnouncementProcessingStatus ProcessingStatus { get; set; } = AnnouncementProcessingStatus.Pending;

    public double[]? Vector { get; set; }

    public int ProcessingRetryCount { get; set; }

    public DateTimeOffset? VectorizedOn { get; set; }

    public int ReporterUserId { get; set; }

    public ApplicationUser ReporterUser { get; set; } = null!;

    public int PetId { get; set; }

    public Pet Pet { get; set; } = null!;

    public ICollection<Comment> Comments { get; set; } = new List<Comment>();

    public ICollection<MatchResult> MatchResultsAsLost { get; set; } = new List<MatchResult>();

    public ICollection<MatchResult> MatchResultsAsFound { get; set; } = new List<MatchResult>();
}

File: mobile-app/types/announcement.ts
import type {
  OnboardingPetAgeCategory,
  OnboardingPetSex,
  OnboardingPetSize,
  OnboardingPetType,
} from "@/types/onboarding";

export type AnnouncementType = "lost" | "found";

export interface AnnouncementDetails {
  city: string;
  country: string;
  dateLastSeen: string;
  description: string;
  showPhone: boolean;
  showTelegram: boolean;
  timeApproximate: string;
}

export enum AnnouncementPetStatus {
  Lost = 0,
  Found = 1,
}

export interface CreateAnnouncementRequest {
  approximateTime: string;
  breed?: string;
  chipNumber?: string;
  city: string;
  country: string;
  isPhonePublic: boolean;
  isTelegramActive: boolean;
  lastDateWhenSeen: string;
  nearLandmark: string;
  petAgeCategory?: OnboardingPetAgeCategory;
  petDetails: string;
  petId?: number;
  petName?: string;
  petPhotoUrl?: string;
  petSex?: OnboardingPetSex;
  petSize?: OnboardingPetSize;
  petStatus: AnnouncementPetStatus;
  petType?: OnboardingPetType;
}

export interface AnnouncementResponse {
  message: string;
  data: {
    id: number;
    petId: number;
    petStatus: number;
    petStatusLabel: string;
    country: string;
    city: string;
    lastDateWhenSeen: string;
    approximateTime: string;
    petDetails: string;
    isPhonePublic: boolean;
    isTelegramActive: boolean;
    nearLandmark: string;
    lastSeenLatitude: number;
    lastSeenLongitude: number;
    isActive: boolean;
    createdOn: string;
    pet: {
      id: number;
      petName: string;
      petType: number;
      petTypeLabel: string;
      petSex: number;
      petSexLabel: string;
      petSize: number;
      petSizeLabel: string;
      petAgeCategory: number;
      petAgeCategoryLabel: string;
      breed: string;
      chipNumber: string;
      description: string;
      petPhotoUrl: string;
    }
  }
}

export type AnnouncementQueryFilter = {
  search?: string;
  petStatus?: number;
  petType?: number;
  city?: string;
  country?: string;
  createdFrom?: string;
  createdTo?: string;
  isActive?: boolean;
  sortBy?: string;
  sortDirection?: 'asc' | 'desc';
  pageNumber?: number;
  pageSize?: number;
};

export type Announcement = {
  id: number;
  petId: number;
  petStatus: number;
  petStatusLabel: string;
  petName: string;
  petType: number;
  petTypeLabel: string;
  breed: string | null;
  petSex: number;
  petSexLabel: string;
  petSize: number;
  petSizeLabel: string;
  petPhotoUrl: string | null;
  country: string;
  city: string;
  nearLandmark: string;
  lastDateWhenSeen: string;
  approximateTime: string;
  petDetails: string;
  isPhonePublic: boolean;
  isTelegramActive: boolean;
  lastSeenLatitude: number;
  lastSeenLongitude: number;
  isActive: boolean;
  createdOn: string;
  commentsCount: number;
}

export type AnnouncementQueryFilterResponse = {
  message: string;
  data: {
    currentPage: number;
    totalPages: number;
    pageSize: number;
    totalCount: number;
    items: Announcement[]
  }
}